zerver/lib/onboarding.py

```diff
@@ -50,6 +50,10 @@ def setup_initial_private_stream(user):
     # type: (UserProfile) -> None
     stream, _ = create_stream_if_needed(user.realm, "core team", invite_only=True,
                                         stream_description="A private stream for core team members.")
+
+    user.realm.signup_notifications_stream = stream
```

Comment:
Let's move this to do_create_realm, next to where realm.notifications_stream is set (the move can be done in a separate commit, at the beginning).

I would do the following in do_create_realm:
    signup_notifications_stream, _ = create_stream_if_needed(realm, Realm.XXX, invite_only=True)
(and then update realm appropriately)

And then this function can just set the description and subscribe the user.